nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/wag/InvokeAWSGatewayApi.java

```diff
@@ -380,4 +355,87 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             }
         }
     }
+
+    @Override
+    public List<ConfigVerificationResult> verify(final ProcessContext context, final ComponentLog verificationLogger, final Map<String, String> attributes) {
+        final List<ConfigVerificationResult> results = new ArrayList<>(super.verify(context, verificationLogger, attributes));
+
+        final String method = context.getProperty(PROP_METHOD).getValue();
+        final String endpoint = context.getProperty(PROP_AWS_GATEWAY_API_ENDPOINT).getValue();
+        final String resource = context.getProperty(PROP_RESOURCE_NAME).getValue();
+        try {
+            final GenericApiGatewayClient client = getConfiguration(context).getClient();
+
+            final GatewayResponse gatewayResponse = invokeGateway(client, context, null, null, attributes, verificationLogger);
```

Comment:
An (unexpected) invoke of the gateway seems like it could be problematic for some use cases (e.g. if calling a non-idempotent API) - would it be better/possible to do some sort of "gateway exists" check instead?

Reply:
Great point -- I should add a check and only execute the call if it's a GET.

Reply:
My concern here is still that a `GET` may not be idempotent - can the verification step be turned off?

If my API Gateway (being a GET) results in other things happening downstream that may impact my system, I might not want this being executed until my NiFi Flow has first done other stuff upstream (e.g. inersted data into a datastore that the API Gateway then does something with ... without that data being present then it could cause my system problems)

Reply:
Happy that this isn't an issue any longer (see https://github.com/apache/nifi/pull/5504#discussion_r763169549) - the change to have the Idempotent methods list was definitely worthwhile I think and I'm happy that the triggering of this check will be at the behest of the user that's configuring the processor (and so they should know whether their service can be "pinged" in this manner)